### PR TITLE
Add sidebar, accept new fortune input

### LINKED ISSUE #3: Complete core fortune sidebar functionality
- [x] #8
- [x] #9
- [x] #10
- [x] #23

See [Hi-Fi prototype](https://www.figma.com/file/38Z0erfGU10hFsjoErmDuD/Wireframes-v.1-(fortune-teller)?type=design&node-id=2-2&t=FfQY52hu6GfemhpE-0).

### LINKED ISSUE #8: Create fortune sidebar
8 boxes in a vertical column to display fortunes. The sidebar should have indices 1-8 which can accept text.

See [Hi-Fi prototype](https://www.figma.com/file/38Z0erfGU10hFsjoErmDuD/Wireframes-v.1-(fortune-teller)?type=design&node-id=2-2&t=FfQY52hu6GfemhpE-0).

**NOT IN SCOPE:** Operations, default fortune values, etc.

### LINKED ISSUE #9: Create new fortune submission overlay

### PR DESCRIPTION
**Pull request by AlbertXDing**
Sidebar with eight previously generated prompts. If the user clicks on one of them, a prompt appears where the user can change the prompts.

### CODE REVIEW

File: assets/scripts/main.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
-// main.js
-
 document.addEventListener('DOMContentLoaded', () => {
-    // init actions
+	const buttons = document.querySelectorAll('button');
```
Comment (phentos) [inline assets/scripts/main.js]: are you sure you want to grab every button everywhere?

File: assets/scripts/main.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+			editBox.style.display = 'block';
+
+            // When the save button is clicked, button's text is changed to the input from editBox
+			save.addEventListener('click', () => {
+				button.textContent = userInput.value;
+				editBox.style.display = 'none';
+			});
+
+		});
+	});
```
Comment (phentos) [inline assets/scripts/main.js]: move to their own function(s), this is way too many things for the page load function

for example, create a new function called `activateSidebarHandler` to call from the page load

also bear in mind that the sidebar has nothing to do with the popup to enter a new fortune, so they should be in separate functions (ie, the sidebar listener calls the popup function to get the new value from the user)

File: assets/scripts/main.js [hunk 3]
```diff
@@ -1,5 +1,21 @@
-// main.js
-
 document.addEventListener('DOMContentLoaded', () => {
-    // init actions
+	const buttons = document.querySelectorAll('button');
+    const editBox = document.getElementById('editBox');
+    const save = document.querySelector('.save');
+	const userInput = document.getElementById('userInput');
```
Comment (phentos) [inline assets/scripts/main.js]: descriptive names are really important:
- `editBox` -- this is meant to be the input field for the user to edit a fortune, right? so it should be named to reflect that, something like `userInputFortuneEditBox`
- `save` -- I can't tell what this is meant to save by reading it, we should be able to tell by the name. Perhaps use something like `userInputFortuneSubmitButton`
- `userInput` sorta works, but it would be nice if we could tell what the element is, especially since it's part of the input overlay

File: index.html [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
+	</head>
+	<body>
+		<button class="button1">Outlook not so good</button>
+		<button class="button2">Signs point to yes</button>
+		<button class="button3">Cannot predict now</button>
+		<button class="button4">Reply hazy, try again later</button>
+		<button class="button5">It is certain</button>
+		<button class="button6">Don't Count on it</button>
+		<button class="button7">Better not tell you now</button>
+		<button class="button8">As I see it, yes</button>
```
Comment (phentos) [inline index.html]: these should be part of a custom element / named container /etc rather than floating in `body`

File: index.html [hunk 5]
```diff
@@ -1,12 +1,25 @@
 <!DOCTYPE html>
-<html lang="en">
```
Comment (phentos) [inline index.html:2]: why did you remove `lang="en"`?

File: index.html [hunk 6]
```diff
@@ -1,12 +1,25 @@
...
+		<button class="button8">As I see it, yes</button>
 
-    <body>
-        <!-- lorem ipsum -->
-    </body>
+        <!-- Textbox that prompts user input -->
+		<div id="editBox" style="display: none;">
+			<input type="text" id="userInput" class="text-input" placeholder="Enter your text">
+			<button class = "save">Save</button>
+		</div>
```
Comment (phentos) [inline index.html]: same deal with earlier comment, better names would be good

**Review by phentos — CHANGES_REQUESTED** [review]
Thanks for the initiative porting Figma! I definitely wasn't looking forward to it myself lol.  

Here's my rationale for most of the changes requested: we want to keep things manageable as we add more functionality, so lets keep things as clean and compartmentalized as we can before merging them. With that in mind, we want to minimize side effects (for example, manipulating every button on the page instead of just the ones you need) and actions per function (a function should do 1 or 2 things; if it needs to do more, it should be multiple functions).

**Comment by AlbertXDing** [comment]
Requested changes have been fixed in commit [de41dfc](https://github.com/cse110-sp23-group1/Origami-Fortune-Teller/pull/15/commits/de41dfc2d617d665095f3692a4f813b2fa01dc17)

**Review by phentos — APPROVED** [review]

File: assets/scripts/main.js [hunk 7]
```diff
@@ -1,21 +1,29 @@
...
 
-	 });
-     });
-});
+    editFortuneBox.style.display = 'block';
+
+    saveButton.addEventListener('click', () => {
+        button.textContent = userInput.value;
+        editFortuneBox.style.display = 'none';
+    });
```
Comment (phentos) [inline assets/scripts/main.js:28]: We don't replace `textContent` before or after the user gives a value, meaning every time they edit it will be prefilled with the last thing they submitted. Imo, it should have the fortune they are currently editing instead -- warrants discussion and an ADR @adarsh249

**Review by phentos — APPROVED** [review]
Awesome work! Huge improvement, much more readable, much more expandable. Great job by @AlbertXDing

File: assets/scripts/main.js (hunk 7 above)
Comment (phentos) [reply assets/scripts/main.js:28]: moved to #23
